### PR TITLE
Pipeline, Anastasia Logunova - 22FPL1

### PR DESCRIPTION
**Pull request by logunova-anastasia**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1 +1,9 @@
-
+beautifulsoup4==4.12.0
+digraph==0.1.4
```
Comment (demid5111) [inline requirements.txt:2]: ?

File: lab_6_pipeline/pipeline.py [hunk 2]
```diff
@@ -99,6 +168,14 @@ def analyze(self, texts: list[str]) -> list[StanzaDocument | str]:
...
             list[StanzaDocument | str]: List of documents
         """
+        docs = []
+
+        for text in texts:
+            analyzed_text = self._analyzer(text)
+            conllu_annotation = analyzed_text._.conll_str
+            docs.append(conllu_annotation)
+
+        return docs
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:178]: rework to comprehension

File: lab_6_pipeline/pipeline.py [hunk 3]
```diff
@@ -62,11 +112,17 @@ def __init__(
...
+        self._corpus = corpus_manager
+        self.analyzer = analyzer
 
     def run(self) -> None:
         """
         Perform basic preprocessing and write processed text to files.
         """
+        for article in self._corpus.get_articles().values():
+            to_cleaned(article)
+            article.set_conllu_info(self.analyzer.analyze(split_by_sentence(article.text)))
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:124]: no. we have to pass to analyzer a batch of documents and not batch of sentences. this idea should considerably change the logic of this method

**Review by demid5111 — CHANGES_REQUESTED** [review]
good start. but only 6? raise the bar!

File: requirements.txt (hunk 1 above)
Comment (logunova-anastasia) [reply requirements.txt:2]: removed it

File: lab_6_pipeline/pipeline.py (hunk 2 above)
Comment (logunova-anastasia) [reply lab_6_pipeline/pipeline.py:178]: did it

File: lab_6_pipeline/pipeline.py [hunk 4]
```diff
@@ -27,16 +45,47 @@ def __init__(self, path_to_raw_txt_data: pathlib.Path) -> None:
...
+        if any(file.stat().st_size == 0 for file in (raw_files + meta_files)):
+            raise InconsistentDatasetError
 
     def _scan_dataset(self) -> None:
         """
         Register each dataset entry.
         """
+        for file in list(self.path_to_raw_txt_data.glob("*_raw.txt")):
+            ind = get_article_id_from_filepath(file)
+            self._storage[ind] = from_raw(file, Article(None, ind))
```
[the remark this answers is not in the document]
Comment (logunova-anastasia) [reply lab_6_pipeline/pipeline.py:88]: did it

File: lab_6_pipeline/pipeline.py [hunk 5]
```diff
@@ -27,16 +45,47 @@ def __init__(self, path_to_raw_txt_data: pathlib.Path) -> None:
...
+            raise InconsistentDatasetError
+
+        sorted_raw_files = sorted(raw_files, key=lambda file: get_article_id_from_filepath(file))
+        sorted_meta_files = sorted(meta_files, key=lambda file: get_article_id_from_filepath(file))
+
+        for ind, (raw, meta) in enumerate(zip(sorted_raw_files, sorted_meta_files)):
+            if ind + 1 != get_article_id_from_filepath(raw) or ind + 1 != get_article_id_from_filepath(meta):
+                raise InconsistentDatasetError
+
+        if any(file.stat().st_size == 0 for file in (raw_files + meta_files)):
```
[the remark this answers is not in the document]
Comment (logunova-anastasia) [reply lab_6_pipeline/pipeline.py:79]: changed it

File: lab_6_pipeline/pipeline.py (hunk 3 above)
Comment (logunova-anastasia) [reply lab_6_pipeline/pipeline.py:124]: done

File: lab_6_pipeline/pipeline.py [hunk 6]
```diff
@@ -220,7 +337,7 @@ def _make_graphs(self, doc: CoNLLUDocument) -> list[DiGraph]:
         """
 
     def _add_children(
-        self, graph: DiGraph, subgraph_to_graph: dict, node_id: int, tree_node: TreeNode
+            self, graph: DiGraph, subgraph_to_graph: dict, node_id: int, tree_node: TreeNode
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:340]: discard unrelated changes

File: lab_6_pipeline/pipeline.py [hunk 7]
```diff
@@ -197,7 +314,7 @@ class PatternSearchPipeline(PipelineProtocol):
     """
 
     def __init__(
-        self, corpus_manager: CorpusManager, analyzer: LibraryWrapper, pos: tuple[str, ...]
+            self, corpus_manager: CorpusManager, analyzer: LibraryWrapper, pos: tuple[str, ...]
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:317]: .

File: lab_6_pipeline/pipeline.py [hunk 8]
```diff
@@ -189,6 +299,13 @@ def _count_frequencies(self, article: Article) -> dict[str, int]:
         Returns:
             dict[str, int]: POS frequencies
         """
+        sentences_features = {}
+        for conllu_sentence in self._analyzer.from_conllu(article).sentences:
+            for word in conllu_sentence.words:
+                word_feature = word.to_dict()['upos']
+                sentences_features[word_feature] = str(self._analyzer.from_conllu(
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:306]: you read conllu for each word in each sentence of conllu document. do you feel something strange?

File: lab_6_pipeline/pipeline.py [hunk 9]
```diff
@@ -53,7 +106,7 @@ class TextProcessingPipeline(PipelineProtocol):
     """
 
     def __init__(
-        self, corpus_manager: CorpusManager, analyzer: LibraryWrapper | None = None
+            self, corpus_manager: CorpusManager, analyzer: LibraryWrapper | None = None
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:109]: .

File: lab_6_pipeline/pipeline.py [hunk 10]
```diff
@@ -27,16 +49,46 @@ def __init__(self, path_to_raw_txt_data: pathlib.Path) -> None:
...
+            if ind + 1 != get_article_id_from_filepath(raw) or ind + 1 != \
+                    get_article_id_from_filepath(meta) or any(
+                    file.stat().st_size == 0 for file in (raw_files + meta_files)):
+                raise InconsistentDatasetError
 
     def _scan_dataset(self) -> None:
         """
         Register each dataset entry.
         """
+        self._storage = {get_article_id_from_filepath(file): from_raw(file, Article(
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:89]: could be better formatting:


```py

a = {
  key: value
  for ...
}
``` 

OR if long value expression

```py

a = {
  key: 
  value
  for ...
}
```

**Review by demid5111 — CHANGES_REQUESTED** [review]
progress is fine. CI is red and try for 10

File: lab_6_pipeline/pipeline.py (hunk 6 above)
Comment (logunova-anastasia) [reply lab_6_pipeline/pipeline.py:340]: done

File: lab_6_pipeline/pipeline.py (hunk 7 above)
Comment (logunova-anastasia) [reply lab_6_pipeline/pipeline.py:317]: done

File: lab_6_pipeline/pipeline.py (hunk 9 above)
Comment (logunova-anastasia) [reply lab_6_pipeline/pipeline.py:109]: done

File: lab_6_pipeline/pipeline.py (hunk 10 above)
Comment (logunova-anastasia) [reply lab_6_pipeline/pipeline.py:89]: changed it

File: lab_6_pipeline/pipeline.py (hunk 8 above)
Comment (logunova-anastasia) [reply lab_6_pipeline/pipeline.py]: indeed, fixed it

File: lab_6_pipeline/pipeline.py [hunk 11]
```diff
@@ -189,6 +301,13 @@ def _count_frequencies(self, article: Article) -> dict[str, int]:
         Returns:
             dict[str, int]: POS frequencies
         """
+        sentences_features = {}
+        str_conllu = str(self._analyzer.from_conllu(article).sentences)
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:305]: why do you cast it to str? it is very strange (!)

**Review by demid5111 — CHANGES_REQUESTED** [review]
almost. one comment left

File: lab_6_pipeline/pipeline.py (hunk 11 above)
Comment (logunova-anastasia) [reply lab_6_pipeline/pipeline.py]: changed it
